cloudify_rest_client/inter_deployment_dependencies.py

```diff
@@ -0,0 +1,153 @@
+        self.update(dependency)
+
+    @property
+    def id(self):
+        return self['id']
+
+    @property
+    def dependency_creator(self):
+        return self['dependency_creator']
+
+    @property
+    def source_deployment_id(self):
+        return self['source_deployment_id']
+
+    @property
+    def target_deployment_id(self):
+        return self['target_deployment_id']
+
+    @property
+    def created_at(self):
+        return self['created_at']
+
+
+class InterDeploymentDependencyClient(object):
+    """
+    Cloudify's plugins update management client.
+    """
+
+    def __init__(self, api):
+        self.api = api
+        self._uri_prefix = 'inter-deployment-dependencies'
+        self._wrapper_cls = InterDeploymentDependency
+
+    def _wrap_list(self, response):
+        return ListResponse(
+            [self._wrapper_cls(item) for item in response['items']],
+            response['metadata']
+        )
+
+    def create(self, dependency_creator, source_deployment, target_deployment):
+        """Creates an inter-deployment dependency.
+
+        :param dependency_creator: a string representing the entity that
+         is responsible for this dependency (e.g. an intrinsic function
+         blueprint path, 'node_instances.some_node_instance', etc.).
+        :param source_deployment: source deployment that depends on the target
+         deployment.
+        :param target_deployment: the deployment that the source deployment
+         depends on.
+        :return: an InterDeploymentDependency object.
+        """
+        data = {
+            'dependency_creator': dependency_creator,
+            'source_deployment': source_deployment,
+            'target_deployment': target_deployment
+        }
+        response = self.api.put(
+            '/{self._uri_prefix}'.format(self=self), data=data)
+        return self._wrapper_cls(response)
+
+    def delete(self,
+               dependency_creator,
+               source_deployment,
+               target_deployment,
+               doesnt_exist_ok=False):
+        """Deletes an inter-deployment dependency.
+
+        :param dependency_creator: a string representing the entity that
+         is responsible for this dependency (e.g. an intrinsic function
+         blueprint path, 'node_instances.some_node_instance', etc.).
+        :param source_deployment: source deployment that depends on the target
+         deployment.
+        :param target_deployment: the deployment that the source deployment
+         depends on.
+        :param doesnt_exist_ok: if it's False then no error will be raised if
+         the dependency doesn't exist. Defaults to False.
+        :return: an InterDeploymentDependency object.
+        """
+        data = {
+            'dependency_creator': dependency_creator,
+            'source_deployment': source_deployment,
+            'target_deployment': target_deployment,
+            'doesnt_exist_ok': doesnt_exist_ok
```

Comment:
`doesnt_exist_ok` feels like something that should be handled here, on the client-side, instead of on the server-side

Reply:
would be a try/except, but however we do have the concept of `.get(..., expected_status_code=x)`, but it has to be a single specific status code
perhaps we should change it to allow expecting a list of codes